docs/reference/hook.md

```diff
@@ -1415,3 +1413,31 @@ TBA
 
 Any unit.
 
+
+
+## Hook and action environment variables
+
+### `<environment variable(s) specific to a hook (kind) or action>`
+
+See the documentation for the relevant hook or action.
+
+(generic-environment-variables)=
+### Generic environment variables
+
+* `PATH` is the usual Unix path, prefixed by a directory containing command line tools through which the hooks can interact with Juju.
+* `JUJU_CHARM_DIR` holds the path to the charm directory.
+* `JUJU_HOOK_NAME` holds the name of the currently executing hook.
+* `JUJU_UNIT_NAME` holds the name of the local unit.
+* `JUJU_CONTEXT_ID`, `JUJU_AGENT_SOCKET_NETWORK` and `JUJU_AGENT_SOCKET_ADDRESS` are set (but should not be messed with: the command line tools won't work without them).
```

Comment:
It is better to just explain what they are for, rather than saying "don't mess with them for reasons".
```suggestion
* `JUJU_CONTEXT_ID`, `JUJU_AGENT_SOCKET_NETWORK` and `JUJU_AGENT_SOCKET_ADDRESS` are used by the hook commands to connect to the local agent.
```

Reply:
Updated, thanks!